src/react/Provider.ts

```diff
@@ -13,8 +20,13 @@ type Options = {
   store?: Store
 }
 
+// In React 19, `use` will allow calling within conditionals and loops.
+//  Refs: https://19.react.dev/reference/react/use#reading-context-with-use
```

Comment:
But it doesn't allow calling `useStore` in a condition. This is pretty unfortunate, but can't be helped.

So, I don't think this PR helps. It rather misleads to misusage.
If you want to use `use`, the current recommendation is to use custom context on your end.